dlrover/python/unified/backend/rl/trainer.py

```diff
@@ -198,28 +282,148 @@ class BaseTrainer(ABC):
+
+    async def start(self):
+        # Initialize the elastic client here
+        logger.info("Start job execution.")
+        await self._setup_workloads()
+
+        self._start_execution()
+
+    async def _setup_workloads(self):
+        start = time.time()
+
+        # setup trainer itself
+        self._setup_trainer()
+
+        # setup other workloads
+        for role, group in self._workload_workers.items():
+            logger.info(f"Setting up workload: {role}...")
+            master_addr = await invoke_actor_t(
+                remote_call.get_master_addr, group[0].name
+            )
+
+            envs = {
+                DLWorkloadEnv.MASTER_ADDR: master_addr[0],
+                DLWorkloadEnv.MASTER_PORT: str(master_addr[1]),
+            }
+
+            res = await ActorBatchInvocation(
+                [
+                    invoke_actor_t(
+                        remote_call.setup_rl_workload,
+                        node.name,
+                        env_dict=envs,
+                    )
+                    for i, node in enumerate(group)
+                ]
+            )
+            res.raise_for_errors()
+
+        elapsed = time.time() - start
+        logger.info(
+            f"Finish setup all rl workloads, cost: {elapsed / 1000:.2f}ms"
+        )
+
+    def _setup_trainer(self):
+        all_workers = PrimeMasterApi.get_all_roles()
+        del all_workers[RLRoleType.TRAINER.name]
+        self._workload_workers = all_workers
+
+        for role, actors_info in self._workload_workers.items():
+            if role not in self._actor_handles:
+                self._actor_handles[role] = []
+            if role not in self._actor_metas:
+                clz = get_class_by_module_and_class_name(
+                    actors_info[0].spec.module_name,
+                    actors_info[0].spec.class_name,
+                )
+                self._actor_metas[role] = (
+                    clz,
+                    actors_info[0].spec.resource.accelerator,
+                )
+
+            for actor_info in actors_info:
+                self._actor_handles[role].append(
+                    ray.get_actor(actor_info.name)
+                )
+
         self.__init_role_group_proxy()
+        self._update_stage_force(WorkerStage.READY)
         logger.info(
             f"Trainer initiated with workloads: {self._get_workloads_size()}, "
             f"role actor metas: {self._actor_metas}, "
-            f"config: {config}"
+            f"config: {self.config}"
         )
 
+    def _start_execution(self):
+        self._update_stage_force(WorkerStage.RUNNING, WorkerStage.READY)
+
+        @contextmanager
+        def wrap_run():
+            try:
+                yield
+                self._update_stage_force(
+                    WorkerStage.FINISHED, WorkerStage.RUNNING
+                )
+
+                invocations = ActorBatchInvocation(
+                    [
+                        invoke_actor_t(
```

Comment:
parameter is same, may use `invoke_actors_t`

Reply:
done